crates/op-rbuilder/src/payload_builder.rs

```diff
@@ -101,12 +104,14 @@ impl CustomOpPayloadBuilder {
         flashblocks_ws_url: String,
         chain_block_time: u64,
         flashblock_block_time: u64,
+        supervisor_url: Option<Url>,
```

Comment:
the right way to do builder pattern is not to include this optional attribute in constructor, instead have it as a separate method
```rust
pub fn supervisor_url(mut self, url: Url) -> Self {
    self.supervisor_url = Some(url);
    self
}
```